src/qtism/runtime/expressions/operators/Utils.php

```diff
@@ -331,23 +331,25 @@ public static function lastPregErrorMessage()
      * Prepare an XSD Regular Expression pattern into a PCRE compliant one.
      *
      * @param string $pattern
+     * @param string $pcreFlags
      * @return string
      */
-     public static function prepareXsdPatternForPcre(string $pattern): string
-     {
-         // XML schema always implicitly anchors the entire regular expression
-         // Neither caret (^) nor dollar ($) sign have special meaning so they are
-         // considered as normal characters.
-         // see http://www.regular-expressions.info/xml.html
-         $pattern = self::withoutStringAnchors($pattern);
-         $pattern = self::escapeSymbols($pattern, ['$', '^']);
-         $pattern = self::pregAddDelimiter('^' . $pattern . '$');
-
-         // XSD regexp always case-sensitive (nothing to do), dot matches white-spaces (use PCRE_DOTALL).
-         $pattern .= 's';
-
-         return $pattern;
-     }
+    public static function prepareXsdPatternForPcre(string $pattern, string $pcreFlags = ''): string
+    {
+        // XML schema always implicitly anchors the entire regular expression
+        // Neither caret (^) nor dollar ($) sign have special meaning so they are
+        // considered as normal characters.
+        // see http://www.regular-expressions.info/xml.html
+        $pattern = self::withoutStringAnchors($pattern);
+        $pattern = self::escapeSymbols($pattern, ['$', '^']);
+        $pattern = self::pregAddDelimiter('^' . $pattern . '$');
+
+        // XSD regexp always case-sensitive (nothing to do), dot matches white-spaces (use PCRE_DOTALL).
+        $pattern .= 's';
+        $pattern .= $pcreFlags;
```

Comment:
I believe the `u` flag is better to be applied by default in all the cases. The values the pattern is applied to are unicode after all.

```suggestion
    public static function prepareXsdPatternForPcre(string $pattern, string $pcreFlags = 'su'): string
    {
        // XML schema always implicitly anchors the entire regular expression
        // Neither caret (^) nor dollar ($) sign have special meaning so they are
        // considered as normal characters.
        // see http://www.regular-expressions.info/xml.html
        $pattern = self::withoutStringAnchors($pattern);
        $pattern = self::escapeSymbols($pattern, ['$', '^']);
        $pattern = self::pregAddDelimiter('^' . $pattern . '$');

        // XSD regexp always case-sensitive (nothing to do), dot matches white-spaces (use PCRE_DOTALL).
        $pattern .= $pcreFlags;
```